src/xhydro/pmp.py

```diff
@@ -69,179 +81,178 @@ def keep_highest_values(da, quantile):
+        Used as an upper limit for the frequency analysis.
+    rebuild_time : bool
+        Whether or not to reconstruct a timeseries with the same dimensions as `pw`.
 
     Returns
     -------
     xr.DataArray
         Precipitable water for a 100-year return period.
-        It has the same dimensions as da_pw.
     """
     # Compute max monthly and add a «year» dimension.
-    da_pw_m = da_pw.resample({"time": "ME"}).max()
-    year = da_pw_m.time.dt.year
-    month = da_pw_m.time.dt.month
-    da_pw_m = da_pw_m.assign_coords(
-        year=("time", year.data), month=("time", month.data)
-    )
-    da_pw_m = da_pw_m.set_index(time=("year", "month")).unstack("time")
-    da_pw_m = da_pw_m.rename({"year": "time"}).squeeze()
+    pw_m = pw.resample({"time": "MS"}).max()
+    pw_m = unstack_dates(pw_m, seasons={m: m for m in range(1, 13)}, new_dim="month")
 
     # Fits distribution
-    params = fit(da_pw_m, dist=dist, method=method)
-    pw100_m = parametric_quantile(params, q=1 - 1 / 100).squeeze().rename("pw100")
-
-    pw_mm = da_pw_m.rename("precipitable_water_monthly")
-    pw_mm = pw_mm.rename({"time": "year"})
+    # FIXME: Use xhydro.frequency_analysis.local instead
+    params = fit(pw_m, dist=dist, method=method)
+    pw100_m = parametric_quantile(params, q=1 - 1 / 100).squeeze()
 
     # Add a limit to PW100 to limit maximization factors.
-    pw_mm_mf = (pw_mm.max(dim="year") * (1.0 + mf)).squeeze()
-
-    pw100_m = pw100_m.where(pw100_m < pw_mm_mf, other=pw_mm_mf)
-    pw100_m = pw100_m.expand_dims(dim={"year": np.unique(year.values)})
-    pw100_m = pw100_m.stack(stacked_coords=("month", "year"))
-    date_index = pd.DatetimeIndex(
-        pd.to_datetime(
-            pd.DataFrame(
-                {
-                    "year": pw100_m.year.data,
-                    "month": pw100_m.month.data,
-                    "day": (xr.ones_like(pw100_m.month).data),
-                    "hour": (xr.ones_like(pw100_m.month).data) * 12,
-                }
+    pw100_m = pw100_m.clip(max=(pw_m.max(dim="time") * (1.0 + mf)))
+
+    if rebuild_time:
+        pw100_m = pw100_m.expand_dims(dim={"year": np.unique(pw.time.dt.year)})
+        pw100_m = pw100_m.stack(stacked_coords=("month", "year"))
+        if isinstance(pw.indexes["time"], pd.core.indexes.datetimes.DatetimeIndex):
+            time_coord = pd.DatetimeIndex(
+                pd.to_datetime({"year": pw100_m.year, "month": pw100_m.month, "day": 1})
             )
+        elif isinstance(pw.indexes["time"], xr.coding.cftimeindex.CFTimeIndex):
+            time_coord = [
+                xclim.core.calendar.datetime_classes[pw.time.dt.calendar](y, m, 1)
+                for y, m in zip(
+                    pw100_m.year.values,
+                    pw100_m.month.values,
+                )
+            ]
+        else:
+            raise ValueError("The type of 'time' was not understood.")
```

Comment:
To rebuild the time series we have to include the time because the interpolation creates an empty xrray.


```suggestion
        time_coord = [
            pw.time.sel(time=(pw.time.dt.month == m) & (pw.time.dt.year == y) & (pw.time.dt.day == 1)).values[0]
            for y, m in zip(
                pw100_m.year.values,
                pw100_m.month.values,
            )
        ]   
```